### ISSUE TITLE
Syslog widget priority filtering not working any more

### ISSUE THREAD

**Issue by Talkabout** [librenms/librenms#13457]
### The problem

Hi,

since https://github.com/librenms/librenms/pull/13411 the syslog widget priority filtering is not working any more. With this pull request the logic was changed in a sense that filtering is done for strings instead of numbers, which is not working for the rsyslog template provided in librenms documentation. I have added some comments and a possible solution in the pull request thread, would be nice to get some answers there or here.

Thanks!

@murrant 

### Output of ./validate.php

```txt
no issue here
```

**Comment by murrant** [comment]
Found the problem.  Some systems store strings and others store numbers... very fancy.

**Comment by Talkabout** [comment]
Hi @murrant ,

maybe it makes sense then to check what is stored in DB (selecting one row and verify) and then change the selection inside the widget configuration to also store strings? That way it would work for both cases...

**Comment by murrant** [comment]
@Talkabout widget should always store numbers.

**Comment by murrant** [comment]
Hmm, I suspect storing numbers in the database is an unsupported configuration.

**Comment by Talkabout** [comment]
we are storing numbers for priority and severity, based on the rsyslog template provided in the librenms documentation. So I assume that most of the people using rsyslog will have numbers there...

**Comment by murrant** [comment]
severity and priority are literally the same thing:

https://www.rsyslog.com/doc/master/configuration/properties.html

**Comment by murrant** [comment]
My proposed changes:
Remove the level field.
Store priority as a number.
Handle non-numeric input in syslog.php
change strings to numbers for existing entries.
update docs

One problem: the syslog.php instance won't be updated when LibreNMS is updated.  Need to restart the syslog daemon for that.

**Comment by Talkabout** [comment]
I think the most critical part here is removing the level column, because this needs adjustments of the syslog template. I cannot judge if this is something we can do as manual migration is required. If we wouldn't remove it I think it should still be fine?

**Comment by murrant** [comment]
no, we can just throw away that field from the template, no need to adjust.

**Comment by Talkabout** [comment]
yes, but this is a manual step that needs to be done by the user itself. If he does not do it, the data stored in syslog table will become broken. That is my only concern.

**Comment by Talkabout** [comment]
maybe it is not... in the syslog.php we could simply check the number of fields provided by rsyslog (or any other facility). If this number contains the "level", we will ignore it (remove it from the array). That way even a "wrong" definition will not harm. What do you say?

**Comment by Talkabout** [comment]
> no, we can just throw away that field from the template, no need to adjust.

maybe my previous comment is saying the same as your previous comment :) Sorry if I missunderstood it!

**Comment by Talkabout** [comment]
Hi @murrant ,

any news on that? I still have locally modified code base to assure the widget filtering works.

Thanks!

**Comment by chenyhsu** [comment]
On my instance,  any filter (program, priority) does not works in syslog viewer.
